### PR TITLE
PHPC-2343: Require PHP 8.1

### PR DESCRIPTION
**Pull request by alcaeus**
PHPC-2343

Best viewed commit by commit, as this removes a lot of compatibility logic necessary for older PHP versions.

### CODE REVIEW

File: src/phongo_compat.h [hunk 1]
```diff
@@ -65,31 +65,6 @@
 #define PHONGO_BREAK_INTENTIONALLY_MISSING
 #endif
 
-#if PHP_VERSION_ID >= 80000
```
Comment (alcaeus) [inline src/phongo_compat.h:68]: I've replaced all these macros with their underlying symbols

**Review by jmikola — APPROVED** [review]
Reviewed all commits independently. LGTM.

**Review by GromNaN — APPROVED** [review]
LGTM for the stubs, with a question.

**Review by GromNaN — APPROVED** [review]

File: src/MongoDB/ClientEncryption.stub.php [hunk 2]
```diff
@@ -7,6 +7,7 @@
 
 namespace MongoDB\Driver;
 
+/** @not-serializable */
```
[the remark this answers is not in the document]
Comment (alcaeus) [reply src/MongoDB/ClientEncryption.stub.php:10]: This is used by the gen_stub tool to add the `ZEND_ACC_NOT_SERIALIZABLE` flag to the class definition: https://github.com/php/php-src/blob/2f3224ea4105c3bebb7db0bee0a57e62dc560a66/build/gen_stub.php#L3472..L3474

**Pull request merged**